pkg/providers/securitygroup/securitygroup.go

```diff
@@ -109,14 +109,14 @@ func (p *DefaultProvider) describeSecurityGroups(request *ecs.DescribeSecurityGr
 		output, err := p.ecsapi.DescribeSecurityGroupsWithOptions(request, runtime)
 		if err != nil {
 			return err
-		} else if output.Body == nil || output.Body.SecurityGroups == nil {
+		} else if output.Body == nil || output.Body.SecurityGroups == nil || len(output.Body.SecurityGroups.SecurityGroup) == 0 {
```

Comment:
It seems that it shouldn’t return an error when len = 0(not found).

Reply:
Should return an error, normally, n+1 queries for `len() == 0` should not exist, should be the last `NextToken == ""` on break

Reply:
What if the user’s tag itself does not select a resource?